kyaml/yaml/fns.go

```diff
@@ -424,12 +424,43 @@ func Lookup(path ...string) PathGetter {
 	return PathGetter{Path: path}
 }
 
-// Lookup returns a PathGetter to lookup a field by its path and create it if it doesn't already
+// LookupCreate returns a PathGetter to lookup a field by its path and create it if it doesn't already
 // exist.
 func LookupCreate(kind yaml.Kind, path ...string) PathGetter {
 	return PathGetter{Path: path, Create: kind}
 }
 
+// ConventionalContainerPaths is a list of paths at which containers typically appear in workload APIs.
+// It is intended for use with FirstNodeThatExists.
+var ConventionalContainerPaths = [][]string{
+	// e.g. Deployment, ReplicaSet, DaemonSet, Job, StatefulSet
+	{"spec", "template", "spec", "containers"},
+	// e.g. CronJob
+	{"spec", "jobTemplate", "spec", "template", "spec", "containers"},
+	// e.g. Pod
+	{"spec", "containers"},
+	// e.g. PodTemplate
+	{"template", "spec", "containers"},
+}
+
+// FirstNodeThatExists returns the RNode for the first of the paths that exists in the resource.
+// It returns nil if none of the given paths exist, and returns an error if one is encountered
+// during lookup.
+func FirstNodeThatExists(resource *RNode, paths [][]string) (*RNode, error) {
+	var result *RNode
+	var err error
+	for _, path := range paths {
+		result, err = PathGetter{Path: path}.Filter(resource)
```

Comment:
Not sure if it really makes a difference, but should this use `Lookup` instead? 

e.g. `result, err = resource.Pipe(yaml.Lookup(path...))`?